jsonrpc/filter_manager.go

```diff
@@ -341,8 +366,17 @@ func (f *FilterManager) GetFilterChanges(id string) (string, error) {
 }
 
 func (f *FilterManager) Uninstall(id string) bool {
-	f.lock.Lock()
+	f.filtersLock.Lock()
+
+	removed := f.removeFilterByID(id)
+
+	f.filtersLock.Unlock()
```

Comment:
Why not defer?

Reply:
Fixed for you :) 6f2fe32ceefcf128da94084513b34f02e1eacef0